src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp

```diff
@@ -100,6 +107,23 @@ XBU::verbose("SubCommand option: read mem");
     // set working variable
     device = deviceCollection.front();
 
+    // check option combinations
+    // mutually exclusive options
+    if (!m_inputFile.empty() && !m_fill.empty())
+      throw xrt_core::error("Please specify either '--input' or '--fill'");
+    // size must be specified with fill
+    if (!m_fill.empty() && m_sizeBytes.empty())
+      throw xrt_core::error("Please specify '--size' with '--fill'");
+
+
+    //-- input file
+    if (!m_inputFile.empty() && !boost::filesystem::exists(m_inputFile))
+      throw xrt_core::error((boost::format("Input file does not exist: '%s'") % m_inputFile).str());
+    
+    input_stream = new std::ifstream(m_inputFile, std::ios::binary);
```

Comment:
Is there a reason why `new` was required?
If so, please do wrapp this value in an `unique_ptr` or equivalent.

Reply:
I moved the input_stream declaration into a better area so that this is not needed.

Reply:
I moved the input_stream declaration into a better area so that this is not needed.